packages/twenty-server/src/engine/core-modules/billing/services/billing-subscription-item.service.ts

```diff
@@ -12,18 +12,23 @@ import { BillingSubscriptionItem } from 'src/engine/core-modules/billing/entitie
 import { BillingProductKey } from 'src/engine/core-modules/billing/enums/billing-product-key.enum';
 import { BillingUsageType } from 'src/engine/core-modules/billing/enums/billing-usage-type.enum';
 import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+import { StripeSubscriptionService } from 'src/engine/core-modules/billing/stripe/services/stripe-subscription.service';
 
 @Injectable()
 export class BillingSubscriptionItemService {
   constructor(
     @InjectRepository(BillingSubscriptionItem, 'core')
     private readonly billingSubscriptionItemRepository: Repository<BillingSubscriptionItem>,
     private readonly twentyConfigService: TwentyConfigService,
+    private readonly stripeSubscriptionService: StripeSubscriptionService,
   ) {}
 
-  async getMeteredSubscriptionItemDetails(subscriptionId: string) {
-    const meteredSubscriptionItems =
-      await this.billingSubscriptionItemRepository.find({
+  async updateMeteredSubscriptionItemPrice(
```

Comment:
Let's make sure you think about security here, like for example I shouldn't be able to consume credits on a given months and then switch back to a lower amount and be reimbursed, etc. I can only switch to a higher amount, etc. Maybe it's covered but make sure you double check all edge-cases :))

Reply:
had the same concern at resolver layer but I think this also makes sense at service layer.

It depends if the service is "internal" and is not meant to be direclty exposed to a resolver. Rule of thumb, the more close we are on the API, the safer we are.
For instance, user.service does not expose getAllUsers which would be dangerous but only deleteUser, etc... which precise checks